app/client/src/git/components/OpsModal/TabRelease/TabReleaseView.tsx

```
@@ -21,13 +22,42 @@ const StyledModalFooter = styled(ModalFooter)`
   min-height: 52px;
 `;
 
-function TabRelease() {
+interface TabReleaseProps {
+  fetchPretag: () => void;
+  createReleaseTag: (params: {
+    tag: string;
+    releaseNote: string;
+    commitSHA: string;
+  }) => void;
+  isCreateReleaseTagLoading: boolean;
+  latestCommitSHA: string | null;
+}
+
+function TabReleaseView({
+  createReleaseTag = noop,
+  fetchPretag = noop,
+  isCreateReleaseTagLoading = false,
+  latestCommitSHA = null,
+}: TabReleaseProps) {
   const [releaseVersion, setReleaseVersion] = useState<string | null>(null);
   const [releaseNotes, setReleaseNotes] = useState<string | null>(null);
 
   const isReleaseDisabled = !releaseVersion || !releaseNotes;
 
-  const handleClickOnRelease = useCallback(() => {}, []);
+  useEffect(
+    function fetchPretagOnInitEffect() {
+      fetchPretag();
+    },
+    [fetchPretag],
+  );
+
+  const handleClickOnRelease = useCallback(() => {
+    createReleaseTag({
+      tag: releaseVersion ?? "",
```

Should we check for the presence of `releaseVersion` and then only dispatch this action?

---

But that would silently fail without an error :(
Will be hard to debug

---

We can throw an error but would making this call with empty releaseVersion would also throw an error then we are fine but throwing error early would be pre-emptive